runtime/services/catalog/artifacts/sql/sql.go

```diff
@@ -24,13 +24,16 @@ func init() {
 	artifacts.Register(".sql", &artifact{})
 }
 
-func (r *artifact) DeSerialise(ctx context.Context, filePath, blob string) (*drivers.CatalogEntry, error) {
+func (r *artifact) DeSerialise(ctx context.Context, filePath, blob string, forceMaterialize bool) (*drivers.CatalogEntry, error) {
 	name := fileutil.Stem(filePath)
 	// extract materialize option before sanitizing query as it will remove that comment
 	materialize := parseMaterializationInfo(blob)
 	if materialize == MaterializeInvalid {
 		return nil, errors.New("invalid materialize type")
 	}
+	if forceMaterialize {
```

Comment:
Instead of force materialization how about changing this to defaultMaterialize and only use it when materialize = MaterializeUnspecified ? 

Essentially, you can set a default at a project level, but individual models can still override the default by explicitly setting up a materialization strategy.

Reply:
Good idea, updated the PR